src/ModelContextProtocol/Protocol/Types/ResourceContents.cs

```diff
@@ -19,17 +26,106 @@ public class ResourceContents
     /// </summary>
     [JsonPropertyName("mimeType")]
     public string? MimeType { get; set; }
+}
 
-    /// <summary>
-    /// The text content of the resource.
-    /// </summary>
-    [JsonPropertyName("text")]
-    public string? Text { get; set; }
+/// <summary>
+/// Converter for <see cref="ResourceContents"/>.
+/// </summary>
+[EditorBrowsable(EditorBrowsableState.Never)]
+public class ResourceContentsConverter : JsonConverter<ResourceContents>
```

Comment:
Nit: as I'm doing in https://github.com/modelcontextprotocol/csharp-sdk/pull/145, let's make this a nested type of ResourceContents and just name it Converter.